README.md
# ELS - Enterprise Linux/EuroLinux Sources - libmx
 
## Usage:
  Checkout branch or tag. Use `make sources` to recreate  SOURCE0 archive.


src/ChangeLog
# Generated by Makefile. Do not edit.

commit 32cb5306567558d7d8c0091c10f6df8dbb06f0bf
Author: Thomas Wood <thomas.wood@intel.com>
Date:   Wed Aug 8 15:06:54 2012 +0100

    Update version number and NEWS for 1.4.7

 NEWS         | 14 ++++++++++++++
 configure.ac |  2 +-
 2 files changed, 15 insertions(+), 1 deletion(-)

commit ea4c5bc5f43daef36b89d54fb09a3a2038a9e0a9
Author: Tristan Van Berkom <tristan.van.berkom@gmail.com>
Date:   Thu May 17 17:52:44 2012 -0400

    MxScrollView: Fixed to clip events for clipped child area in ClutterActor->pick()
    
    Furthermore this patch avoids warnings from calling apis on the child
    actor when NULL.

 mx/mx-scroll-view.c | 24 ++++++++++++++++++++----
 1 file changed, 20 insertions(+), 4 deletions(-)

commit c58679bf00a72fe744b1467610ad735d62c0886d
Author: Tristan Van Berkom <tristan.van.berkom@gmail.com>
Date:   Sat Jan 7 15:55:43 2012 +0900

    widget: Handle NULL ClutterEventCrossing->related actor in mx_widget_leave
    
    In the case that the mouse exits the stage, the ->related actor
    argument will be NULL, this needs to be caught before sending
    the NULL actor to clutter_actor_contains() and firing a needless
    assertion failure.

 mx/mx-widget.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

commit 7f590d07fec9cf1bc41d348beab2649be8f355d0
Author: Evan Nemerson <evan@coeus-group.com>
Date:   Tue Jan 24 21:25:57 2012 -0800

    image: add some annotations for mx_image_set_from_buffer/data
    
    Fixes: http://bugzilla.clutter-project.org/show_bug.cgi?id=2688

 mx/mx-image.c | 12 +++++++-----
 1 file changed, 7 insertions(+), 5 deletions(-)

commit 670ab57a91eece6f1b168f803e2e57cfc6b31c1b
Author: Thomas Wood <thomas.wood@intel.com>
Date:   Tue May 8 15:09:59 2012 +0100

    Update version number and NEWS for 1.4.6

 NEWS         | 9 +++++++++
 configure.ac | 2 +-
 2 files changed, 10 insertions(+), 1 deletion(-)

commit 122c45571ebaccecb378f846be96f3525bf72493
Author: Justin Lecher <jlec@gentoo.org>
Date:   Tue Apr 3 08:34:04 2012 +0200

    mx fails to link with ld.gold linker
    
    Due to the extended underlinking detection of gold,
    it is necessary to add libm to the linked libs.
    
    https://github.com/clutter-project/mx/issues/49

 mx/Makefile.am | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

commit 0e318fbe92ecdd9b5174c76c48cd8a5a3e7bf169
Author: Thomas Wood <thomas.wood@intel.com>
Date:   Mon Apr 2 16:00:16 2012 +0100

    stack: don't call mx_stack_paint during pick
    
    Calling mx_stack_paint from the pick virtual function would cause the
    background colour to be painted into the pick buffer.

 mx/mx-stack.c | 21 +++++++++++++++------
 1 file changed, 15 insertions(+), 6 deletions(-)

commit 21c3f3e4c0622ae27ecad902fa5a2fee74e93c1c
Author: Thomas Wood <thomas.wood@intel.com>
Date:   Fri Mar 30 13:53:06 2012 +0100

    Update version number and NEWS for 1.4.5

 NEWS         | 18 ++++++++++++++++++
 configure.ac |  2 +-
 2 files changed, 19 insertions(+), 1 deletion(-)

commit 245fb721514264fdc96f9a29683628db153a4430
Author: Thomas Wood <thomas.wood@intel.com>
Date:   Tue Mar 27 12:10:45 2012 +0100

    dialog: ensure that the initial size of the dialog is correct

 mx/mx-dialog.c | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

commit 1050f1724b1a21b576a74c5ca851eb6c7b3d9fbc
Author: Thomas Wood <thomas.wood@intel.com>
Date:   Tue Mar 27 12:10:28 2012 +0100

    dialog: disable the blurred background when using Clutter ≥ 1.10
    
    The blur effect using MxOffscreen does not work with Clutter 1.10.

 mx/mx-dialog.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

commit fd0271cfe551ce43f1cffdae375e890f6db57f73
Author: Thomas Wood <thomas.wood@intel.com>
Date:   Tue Mar 27 11:37:42 2012 +0100

    css: pseudo-class should be identified by ':' in debug output

 mx/mx-css.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

commit c756357f158c39e0a889e54d9602da565885c4e1
Author: Thomas Wood <thomas.wood@intel.com>
Date:   Tue Mar 27 11:34:50 2012 +0100

    combo-box: don't use NULL when setting the ClutterText "text" property

 mx/mx-combo-box.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

commit b9d70ef880549791e4d07d27b96d729742febe11
Author: Thomas Wood <thomas.wood@intel.com>

src/mx/mx-marshal.h

#ifndef ___mx_marshal_MARSHAL_H__
#define ___mx_marshal_MARSHAL_H__

#include	<glib-object.h>

G_BEGIN_DECLS

/* VOID:OBJECT (./mx-marshal.list:1) */
#define _mx_marshal_VOID__OBJECT	g_cclosure_marshal_VOID__OBJECT

/* VOID:VOID (./mx-marshal.list:2) */
#define _mx_marshal_VOID__VOID	g_cclosure_marshal_VOID__VOID

/* VOID:PARAM (./mx-marshal.list:3) */
#define _mx_marshal_VOID__PARAM	g_cclosure_marshal_VOID__PARAM

/* VOID:POINTER (./mx-marshal.list:4) */
#define _mx_marshal_VOID__POINTER	g_cclosure_marshal_VOID__POINTER

/* VOID:UINT (./mx-marshal.list:5) */
#define _mx_marshal_VOID__UINT	g_cclosure_marshal_VOID__UINT

/* VOID:ULONG (./mx-marshal.list:6) */
#define _mx_marshal_VOID__ULONG	g_cclosure_marshal_VOID__ULONG

/* VOID:FLAGS (./mx-marshal.list:7) */
#define _mx_marshal_VOID__FLAGS	g_cclosure_marshal_VOID__FLAGS

/* VOID:BOXED (./mx-marshal.list:8) */
#define _mx_marshal_VOID__BOXED	g_cclosure_marshal_VOID__BOXED

/* VOID:UINT,UINT (./mx-marshal.list:9) */
extern void _mx_marshal_VOID__UINT_UINT (GClosure     *closure,
                                         GValue       *return_value,
                                         guint         n_param_values,
                                         const GValue *param_values,
                                         gpointer      invocation_hint,
                                         gpointer      marshal_data);

/* VOID:UINT,OBJECT (./mx-marshal.list:10) */
extern void _mx_marshal_VOID__UINT_OBJECT (GClosure     *closure,
                                           GValue       *return_value,
                                           guint         n_param_values,
                                           const GValue *param_values,
                                           gpointer      invocation_hint,
                                           gpointer      marshal_data);

/* VOID:ULONG,BOXED (./mx-marshal.list:11) */
extern void _mx_marshal_VOID__ULONG_BOXED (GClosure     *closure,
                                           GValue       *return_value,
                                           guint         n_param_values,
                                           const GValue *param_values,
                                           gpointer      invocation_hint,
                                           gpointer      marshal_data);

/* VOID:ULONG,OBJECT (./mx-marshal.list:12) */
extern void _mx_marshal_VOID__ULONG_OBJECT (GClosure     *closure,
                                            GValue       *return_value,
                                            guint         n_param_values,
                                            const GValue *param_values,
                                            gpointer      invocation_hint,
                                            gpointer      marshal_data);

/* VOID:ULONG,OBJECT,OBJECT (./mx-marshal.list:13) */
extern void _mx_marshal_VOID__ULONG_OBJECT_OBJECT (GClosure     *closure,
                                                   GValue       *return_value,
                                                   guint         n_param_values,
                                                   const GValue *param_values,
                                                   gpointer      invocation_hint,
                                                   gpointer      marshal_data);

/* VOID:OBJECT,OBJECT (./mx-marshal.list:14) */
extern void _mx_marshal_VOID__OBJECT_OBJECT (GClosure     *closure,
                                             GValue       *return_value,
                                             guint         n_param_values,
                                             const GValue *param_values,
                                             gpointer      invocation_hint,
                                             gpointer      marshal_data);

/* VOID:STRING,OBJECT (./mx-marshal.list:15) */
extern void _mx_marshal_VOID__STRING_OBJECT (GClosure     *closure,
                                             GValue       *return_value,
                                             guint         n_param_values,
                                             const GValue *param_values,
                                             gpointer      invocation_hint,
                                             gpointer      marshal_data);

/* VOID:OBJECT,OBJECT,INT,INT (./mx-marshal.list:16) */
extern void _mx_marshal_VOID__OBJECT_OBJECT_INT_INT (GClosure     *closure,
                                                     GValue       *return_value,
                                                     guint         n_param_values,
                                                     const GValue *param_values,
                                                     gpointer      invocation_hint,
                                                     gpointer      marshal_data);

/* VOID:OBJECT,FLOAT,FLOAT,INT,ENUM (./mx-marshal.list:17) */
extern void _mx_marshal_VOID__OBJECT_FLOAT_FLOAT_INT_ENUM (GClosure     *closure,
                                                           GValue       *return_value,
                                                           guint         n_param_values,
                                                           const GValue *param_values,
                                                           gpointer      invocation_hint,
                                                           gpointer      marshal_data);

/* VOID:FLOAT,FLOAT,INT,ENUM (./mx-marshal.list:18) */
extern void _mx_marshal_VOID__FLOAT_FLOAT_INT_ENUM (GClosure     *closure,
                                                    GValue       *return_value,
                                                    guint         n_param_values,
                                                    const GValue *param_values,
                                                    gpointer      invocation_hint,
                                                    gpointer      marshal_data);

/* VOID:FLOAT,FLOAT (./mx-marshal.list:19) */
extern void _mx_marshal_VOID__FLOAT_FLOAT (GClosure     *closure,
                                           GValue       *return_value,
                                           guint         n_param_values,
                                           const GValue *param_values,
                                           gpointer      invocation_hint,
                                           gpointer      marshal_data);

/* BOOL:FLOAT,FLOAT,ENUM (./mx-marshal.list:20) */
extern void _mx_marshal_BOOLEAN__FLOAT_FLOAT_ENUM (GClosure     *closure,
                                                   GValue       *return_value,
                                                   guint         n_param_values,
                                                   const GValue *param_values,
                                                   gpointer      invocation_hint,
                                                   gpointer      marshal_data);
#define _mx_marshal_BOOL__FLOAT_FLOAT_ENUM	_mx_marshal_BOOLEAN__FLOAT_FLOAT_ENUM

/* BOOL:VOID (./mx-marshal.list:21) */
extern void _mx_marshal_BOOLEAN__VOID (GClosure     *closure,
                                       GValue       *return_value,
                                       guint         n_param_values,
                                       const GValue *param_values,
                                       gpointer      invocation_hint,
                                       gpointer      marshal_data);
#define _mx_marshal_BOOL__VOID	_mx_marshal_BOOLEAN__VOID

G_END_DECLS

#endif /* ___mx_marshal_MARSHAL_H__ */



src/mx/mx-marshal.c
#include "mx-marshal.h"

#include	<glib-object.h>


#ifdef G_ENABLE_DEBUG
#define g_marshal_value_peek_boolean(v)  g_value_get_boolean (v)
#define g_marshal_value_peek_char(v)     g_value_get_schar (v)
#define g_marshal_value_peek_uchar(v)    g_value_get_uchar (v)
#define g_marshal_value_peek_int(v)      g_value_get_int (v)
#define g_marshal_value_peek_uint(v)     g_value_get_uint (v)
#define g_marshal_value_peek_long(v)     g_value_get_long (v)
#define g_marshal_value_peek_ulong(v)    g_value_get_ulong (v)
#define g_marshal_value_peek_int64(v)    g_value_get_int64 (v)
#define g_marshal_value_peek_uint64(v)   g_value_get_uint64 (v)
#define g_marshal_value_peek_enum(v)     g_value_get_enum (v)
#define g_marshal_value_peek_flags(v)    g_value_get_flags (v)
#define g_marshal_value_peek_float(v)    g_value_get_float (v)
#define g_marshal_value_peek_double(v)   g_value_get_double (v)
#define g_marshal_value_peek_string(v)   (char*) g_value_get_string (v)
#define g_marshal_value_peek_param(v)    g_value_get_param (v)
#define g_marshal_value_peek_boxed(v)    g_value_get_boxed (v)
#define g_marshal_value_peek_pointer(v)  g_value_get_pointer (v)
#define g_marshal_value_peek_object(v)   g_value_get_object (v)
#define g_marshal_value_peek_variant(v)  g_value_get_variant (v)
#else /* !G_ENABLE_DEBUG */
/* WARNING: This code accesses GValues directly, which is UNSUPPORTED API.
 *          Do not access GValues directly in your code. Instead, use the
 *          g_value_get_*() functions
 */
#define g_marshal_value_peek_boolean(v)  (v)->data[0].v_int
#define g_marshal_value_peek_char(v)     (v)->data[0].v_int
#define g_marshal_value_peek_uchar(v)    (v)->data[0].v_uint
#define g_marshal_value_peek_int(v)      (v)->data[0].v_int
#define g_marshal_value_peek_uint(v)     (v)->data[0].v_uint
#define g_marshal_value_peek_long(v)     (v)->data[0].v_long
#define g_marshal_value_peek_ulong(v)    (v)->data[0].v_ulong
#define g_marshal_value_peek_int64(v)    (v)->data[0].v_int64
#define g_marshal_value_peek_uint64(v)   (v)->data[0].v_uint64
#define g_marshal_value_peek_enum(v)     (v)->data[0].v_long
#define g_marshal_value_peek_flags(v)    (v)->data[0].v_ulong
#define g_marshal_value_peek_float(v)    (v)->data[0].v_float
#define g_marshal_value_peek_double(v)   (v)->data[0].v_double
#define g_marshal_value_peek_string(v)   (v)->data[0].v_pointer
#define g_marshal_value_peek_param(v)    (v)->data[0].v_pointer
#define g_marshal_value_peek_boxed(v)    (v)->data[0].v_pointer
#define g_marshal_value_peek_pointer(v)  (v)->data[0].v_pointer
#define g_marshal_value_peek_object(v)   (v)->data[0].v_pointer
#define g_marshal_value_peek_variant(v)  (v)->data[0].v_pointer
#endif /* !G_ENABLE_DEBUG */


/* VOID:OBJECT (./mx-marshal.list:1) */

/* VOID:VOID (./mx-marshal.list:2) */

/* VOID:PARAM (./mx-marshal.list:3) */

/* VOID:POINTER (./mx-marshal.list:4) */

/* VOID:UINT (./mx-marshal.list:5) */

/* VOID:ULONG (./mx-marshal.list:6) */

/* VOID:FLAGS (./mx-marshal.list:7) */

/* VOID:BOXED (./mx-marshal.list:8) */

/* VOID:UINT,UINT (./mx-marshal.list:9) */
void
_mx_marshal_VOID__UINT_UINT (GClosure     *closure,
                             GValue       *return_value G_GNUC_UNUSED,
                             guint         n_param_values,
                             const GValue *param_values,
                             gpointer      invocation_hint G_GNUC_UNUSED,
                             gpointer      marshal_data)
{
  typedef void (*GMarshalFunc_VOID__UINT_UINT) (gpointer     data1,
                                                guint        arg_1,
                                                guint        arg_2,
                                                gpointer     data2);
  register GMarshalFunc_VOID__UINT_UINT callback;
  register GCClosure *cc = (GCClosure*) closure;
  register gpointer data1, data2;

  g_return_if_fail (n_param_values == 3);

  if (G_CCLOSURE_SWAP_DATA (closure))
    {
      data1 = closure->data;
      data2 = g_value_peek_pointer (param_values + 0);
    }
  else
    {
      data1 = g_value_peek_pointer (param_values + 0);
      data2 = closure->data;
    }
  callback = (GMarshalFunc_VOID__UINT_UINT) (marshal_data ? marshal_data : cc->callback);

  callback (data1,
            g_marshal_value_peek_uint (param_values + 1),
            g_marshal_value_peek_uint (param_values + 2),
            data2);
}

/* VOID:UINT,OBJECT (./mx-marshal.list:10) */
void
_mx_marshal_VOID__UINT_OBJECT (GClosure     *closure,
                               GValue       *return_value G_GNUC_UNUSED,
                               guint         n_param_values,
                               const GValue *param_values,
                               gpointer      invocation_hint G_GNUC_UNUSED,
                               gpointer      marshal_data)
{
  typedef void (*GMarshalFunc_VOID__UINT_OBJECT) (gpointer     data1,
                                                  guint        arg_1,
                                                  gpointer     arg_2,
                                                  gpointer     data2);
  register GMarshalFunc_VOID__UINT_OBJECT callback;
  register GCClosure *cc = (GCClosure*) closure;
  register gpointer data1, data2;

  g_return_if_fail (n_param_values == 3);

  if (G_CCLOSURE_SWAP_DATA (closure))
    {
      data1 = closure->data;
      data2 = g_value_peek_pointer (param_values + 0);
    }
  else
    {
      data1 = g_value_peek_pointer (param_values + 0);
      data2 = closure->data;
    }
  callback = (GMarshalFunc_VOID__UINT_OBJECT) (marshal_data ? marshal_data : cc->callback);

  callback (data1,
            g_marshal_value_peek_uint (param_values + 1),
            g_marshal_value_peek_object (param_values + 2),
            data2);
}

/* VOID:ULONG,BOXED (./mx-marshal.list:11) */
void
_mx_marshal_VOID__ULONG_BOXED (GClosure     *closure,
                               GValue       *return_value G_GNUC_UNUSED,
                               guint         n_param_values,
                               const GValue *param_values,
                               gpointer      invocation_hint G_GNUC_UNUSED,
                               gpointer      marshal_data)
{
  typedef void (*GMarshalFunc_VOID__ULONG_BOXED) (gpointer     data1,
                                                  gulong       arg_1,
                                                  gpointer     arg_2,
                                                  gpointer     data2);
  register GMarshalFunc_VOID__ULONG_BOXED callback;
  register GCClosure *cc = (GCClosure*) closure;
  register gpointer data1, data2;

  g_return_if_fail (n_param_values == 3);

  if (G_CCLOSURE_SWAP_DATA (closure))
    {
      data1 = closure->data;
      data2 = g_value_peek_pointer (param_values + 0);
    }
  else
    {
      data1 = g_value_peek_pointer (param_values + 0);
      data2 = closure->data;
    }
  callback = (GMarshalFunc_VOID__ULONG_BOXED) (marshal_data ? marshal_data : cc->callback);

  callback (data1,
            g_marshal_value_peek_ulong (param_values + 1),
            g_marshal_value_peek_boxed (param_values + 2),
            data2);
}

/* VOID:ULONG,OBJECT (./mx-marshal.list:12) */
void
_mx_marshal_VOID__ULONG_OBJECT (GClosure     *closure,
                                GValue       *return_value G_GNUC_UNUSED,
                                guint         n_param_values,
                                const GValue *param_values,
                                gpointer      invocation_hint G_GNUC_UNUSED,
                                gpointer      marshal_data)
{
  typedef void (*GMarshalFunc_VOID__ULONG_OBJECT) (gpointer     data1,
                                                   gulong       arg_1,
                                                   gpointer     arg_2,
                                                   gpointer     data2);
  register GMarshalFunc_VOID__ULONG_OBJECT callback;
  register GCClosure *cc = (GCClosure*) closure;
  register gpointer data1, data2;

  g_return_if_fail (n_param_values == 3);

  if (G_CCLOSURE_SWAP_DATA (closure))
    {
      data1 = closure->data;
      data2 = g_value_peek_pointer (param_values + 0);
    }
  else
    {
      data1 = g_value_peek_pointer (param_values + 0);
      data2 = closure->data;
    }
  callback = (GMarshalFunc_VOID__ULONG_OBJECT) (marshal_data ? marshal_data : cc->callback);

  callback (data1,
            g_marshal_value_peek_ulong (param_values + 1),
            g_marshal_value_peek_object (param_values + 2),
            data2);
}

/* VOID:ULONG,OBJECT,OBJECT (./mx-marshal.list:13) */
void
_mx_marshal_VOID__ULONG_OBJECT_OBJECT (GClosure     *closure,
                                       GValue       *return_value G_GNUC_UNUSED,
                                       guint         n_param_values,
                                       const GValue *param_values,
                                       gpointer      invocation_hint G_GNUC_UNUSED,
                                       gpointer      marshal_data)
{
  typedef void (*GMarshalFunc_VOID__ULONG_OBJECT_OBJECT) (gpointer     data1,
                                                          gulong       arg_1,
                                                          gpointer     arg_2,
                                                          gpointer     arg_3,
                                                          gpointer     data2);
  register GMarshalFunc_VOID__ULONG_OBJECT_OBJECT callback;
  register GCClosure *cc = (GCClosure*) closure;
  register gpointer data1, data2;

  g_return_if_fail (n_param_values == 4);

  if (G_CCLOSURE_SWAP_DATA (closure))
    {
      data1 = closure->data;
      data2 = g_value_peek_pointer (param_values + 0);
    }
  else
    {
      data1 = g_value_peek_pointer (param_values + 0);
      data2 = closure->data;
    }
  callback = (GMarshalFunc_VOID__ULONG_OBJECT_OBJECT) (marshal_data ? marshal_data : cc->callback);

  callback (data1,
            g_marshal_value_peek_ulong (param_values + 1),
            g_marshal_value_peek_object (param_values + 2),
            g_marshal_value_peek_object (param_values + 3),
            data2);
}

/* VOID:OBJECT,OBJECT (./mx-marshal.list:14) */
void
_mx_marshal_VOID__OBJECT_OBJECT (GClosure     *closure,
                                 GValue       *return_value G_GNUC_UNUSED,
                                 guint         n_param_values,
                                 const GValue *param_values,
                                 gpointer      invocation_hint G_GNUC_UNUSED,
                                 gpointer      marshal_data)
{
  typedef void (*GMarshalFunc_VOID__OBJECT_OBJECT) (gpointer     data1,
                                                    gpointer     arg_1,
                                                    gpointer     arg_2,
                                                    gpointer     data2);
  register GMarshalFunc_VOID__OBJECT_OBJECT callback;
  register GCClosure *cc = (GCClosure*) closure;
  register gpointer data1, data2;

  g_return_if_fail (n_param_values == 3);

  if (G_CCLOSURE_SWAP_DATA (closure))
    {
      data1 = closure->data;
      data2 = g_value_peek_pointer (param_values + 0);
    }
  else
    {
      data1 = g_value_peek_pointer (param_values + 0);
      data2 = closure->data;
    }
  callback = (GMarshalFunc_VOID__OBJECT_OBJECT) (marshal_data ? marshal_data : cc->callback);

  callback (data1,
            g_marshal_value_peek_object (param_values + 1),
            g_marshal_value_peek_object (param_values + 2),
            data2);
}

/* VOID:STRING,OBJECT (./mx-marshal.list:15) */
void
_mx_marshal_VOID__STRING_OBJECT (GClosure     *closure,
                                 GValue       *return_value G_GNUC_UNUSED,
                                 guint         n_param_values,
                                 const GValue *param_values,
                                 gpointer      invocation_hint G_GNUC_UNUSED,
                                 gpointer      marshal_data)
{
  typedef void (*GMarshalFunc_VOID__STRING_OBJECT) (gpointer     data1,
                                                    gpointer     arg_1,
                                                    gpointer     arg_2,
                                                    gpointer     data2);
  register GMarshalFunc_VOID__STRING_OBJECT callback;
  register GCClosure *cc = (GCClosure*) closure;
  register gpointer data1, data2;

  g_return_if_fail (n_param_values == 3);

  if (G_CCLOSURE_SWAP_DATA (closure))
    {
      data1 = closure->data;
      data2 = g_value_peek_pointer (param_values + 0);
    }
  else
    {
      data1 = g_value_peek_pointer (param_values + 0);
      data2 = closure->data;
    }
  callback = (GMarshalFunc_VOID__STRING_OBJECT) (marshal_data ? marshal_data : cc->callback);

  callback (data1,
            g_marshal_value_peek_string (param_values + 1),
            g_marshal_value_peek_object (param_values + 2),
            data2);
}

/* VOID:OBJECT,OBJECT,INT,INT (./mx-marshal.list:16) */
void
_mx_marshal_VOID__OBJECT_OBJECT_INT_INT (GClosure     *closure,
                                         GValue       *return_value G_GNUC_UNUSED,
                                         guint         n_param_values,
                                         const GValue *param_values,
                                         gpointer      invocation_hint G_GNUC_UNUSED,
                                         gpointer      marshal_data)
{
  typedef void (*GMarshalFunc_VOID__OBJECT_OBJECT_INT_INT) (gpointer     data1,
                                                            gpointer     arg_1,
                                                            gpointer     arg_2,
                                                            gint         arg_3,
                                                            gint         arg_4,
                                                            gpointer     data2);
  register GMarshalFunc_VOID__OBJECT_OBJECT_INT_INT callback;
  register GCClosure *cc = (GCClosure*) closure;
  register gpointer data1, data2;

  g_return_if_fail (n_param_values == 5);

  if (G_CCLOSURE_SWAP_DATA (closure))
    {
      data1 = closure->data;
      data2 = g_value_peek_pointer (param_values + 0);
    }
  else
    {
      data1 = g_value_peek_pointer (param_values + 0);
      data2 = closure->data;
    }
  callback = (GMarshalFunc_VOID__OBJECT_OBJECT_INT_INT) (marshal_data ? marshal_data : cc->callback);

  callback (data1,
            g_marshal_value_peek_object (param_values + 1),
            g_marshal_value_peek_object (param_values + 2),
            g_marshal_value_peek_int (param_values + 3),
            g_marshal_value_peek_int (param_values + 4),
            data2);
}

/* VOID:OBJECT,FLOAT,FLOAT,INT,ENUM (./mx-marshal.list:17) */
void
_mx_marshal_VOID__OBJECT_FLOAT_FLOAT_INT_ENUM (GClosure     *closure,
                                               GValue       *return_value G_GNUC_UNUSED,
                                               guint         n_param_values,
                                               const GValue *param_values,
                                               gpointer      invocation_hint G_GNUC_UNUSED,
                                               gpointer      marshal_data)
{
  typedef void (*GMarshalFunc_VOID__OBJECT_FLOAT_FLOAT_INT_ENUM) (gpointer     data1,
                                                                  gpointer     arg_1,
                                                                  gfloat       arg_2,
                                                                  gfloat       arg_3,
                                                                  gint         arg_4,
                                                                  gint         arg_5,
                                                                  gpointer     data2);
  register GMarshalFunc_VOID__OBJECT_FLOAT_FLOAT_INT_ENUM callback;
  register GCClosure *cc = (GCClosure*) closure;
  register gpointer data1, data2;

  g_return_if_fail (n_param_values == 6);

  if (G_CCLOSURE_SWAP_DATA (closure))
    {
      data1 = closure->data;
      data2 = g_value_peek_pointer (param_values + 0);
    }
  else
    {
      data1 = g_value_peek_pointer (param_values + 0);
      data2 = closure->data;
    }
  callback = (GMarshalFunc_VOID__OBJECT_FLOAT_FLOAT_INT_ENUM) (marshal_data ? marshal_data : cc->callback);

  callback (data1,
            g_marshal_value_peek_object (param_values + 1),
            g_marshal_value_peek_float (param_values + 2),
            g_marshal_value_peek_float (param_values + 3),
            g_marshal_value_peek_int (param_values + 4),
            g_marshal_value_peek_enum (param_values + 5),
            data2);
}

/* VOID:FLOAT,FLOAT,INT,ENUM (./mx-marshal.list:18) */
void
_mx_marshal_VOID__FLOAT_FLOAT_INT_ENUM (GClosure     *closure,
                                        GValue       *return_value G_GNUC_UNUSED,
                                        guint         n_param_values,
                                        const GValue *param_values,
                                        gpointer      invocation_hint G_GNUC_UNUSED,
                                        gpointer      marshal_data)
{
  typedef void (*GMarshalFunc_VOID__FLOAT_FLOAT_INT_ENUM) (gpointer     data1,
                                                           gfloat       arg_1,
                                                           gfloat       arg_2,
                                                           gint         arg_3,
                                                           gint         arg_4,
                                                           gpointer     data2);
  register GMarshalFunc_VOID__FLOAT_FLOAT_INT_ENUM callback;
  register GCClosure *cc = (GCClosure*) closure;
  register gpointer data1, data2;

  g_return_if_fail (n_param_values == 5);

  if (G_CCLOSURE_SWAP_DATA (closure))
    {
      data1 = closure->data;
      data2 = g_value_peek_pointer (param_values + 0);
    }
  else
    {
      data1 = g_value_peek_pointer (param_values + 0);
      data2 = closure->data;
    }
  callback = (GMarshalFunc_VOID__FLOAT_FLOAT_INT_ENUM) (marshal_data ? marshal_data : cc->callback);

  callback (data1,
            g_marshal_value_peek_float (param_values + 1),
            g_marshal_value_peek_float (param_values + 2),
            g_marshal_value_peek_int (param_values + 3),
            g_marshal_value_peek_enum (param_values + 4),
            data2);
}

/* VOID:FLOAT,FLOAT (./mx-marshal.list:19) */
void
_mx_marshal_VOID__FLOAT_FLOAT (GClosure     *closure,
                               GValue       *return_value G_GNUC_UNUSED,
                               guint         n_param_values,
                               const GValue *param_values,
                               gpointer      invocation_hint G_GNUC_UNUSED,
                               gpointer      marshal_data)
{
  typedef void (*GMarshalFunc_VOID__FLOAT_FLOAT) (gpointer     data1,
                                                  gfloat       arg_1,
                                                  gfloat       arg_2,
                                                  gpointer     data2);
  register GMarshalFunc_VOID__FLOAT_FLOAT callback;
  register GCClosure *cc = (GCClosure*) closure;
  register gpointer data1, data2;

  g_return_if_fail (n_param_values == 3);

  if (G_CCLOSURE_SWAP_DATA (closure))
    {
      data1 = closure->data;
      data2 = g_value_peek_pointer (param_values + 0);
    }
  else
    {
      data1 = g_value_peek_pointer (param_values + 0);
      data2 = closure->data;
    }
  callback = (GMarshalFunc_VOID__FLOAT_FLOAT) (marshal_data ? marshal_data : cc->callback);

  callback (data1,
            g_marshal_value_peek_float (param_values + 1),
            g_marshal_value_peek_float (param_values + 2),
            data2);
}

/* BOOL:FLOAT,FLOAT,ENUM (./mx-marshal.list:20) */
void
_mx_marshal_BOOLEAN__FLOAT_FLOAT_ENUM (GClosure     *closure,
                                       GValue       *return_value G_GNUC_UNUSED,
                                       guint         n_param_values,
                                       const GValue *param_values,
                                       gpointer      invocation_hint G_GNUC_UNUSED,
                                       gpointer      marshal_data)
{
  typedef gboolean (*GMarshalFunc_BOOLEAN__FLOAT_FLOAT_ENUM) (gpointer     data1,
                                                              gfloat       arg_1,
                                                              gfloat       arg_2,
                                                              gint         arg_3,
                                                              gpointer     data2);
  register GMarshalFunc_BOOLEAN__FLOAT_FLOAT_ENUM callback;
  register GCClosure *cc = (GCClosure*) closure;
  register gpointer data1, data2;
  gboolean v_return;

  g_return_if_fail (return_value != NULL);
  g_return_if_fail (n_param_values == 4);

  if (G_CCLOSURE_SWAP_DATA (closure))
    {
      data1 = closure->data;
      data2 = g_value_peek_pointer (param_values + 0);
    }
  else
    {
      data1 = g_value_peek_pointer (param_values + 0);
      data2 = closure->data;
    }
  callback = (GMarshalFunc_BOOLEAN__FLOAT_FLOAT_ENUM) (marshal_data ? marshal_data : cc->callback);

  v_return = callback (data1,
                       g_marshal_value_peek_float (param_values + 1),
                       g_marshal_value_peek_float (param_values + 2),
                       g_marshal_value_peek_enum (param_values + 3),
                       data2);

  g_value_set_boolean (return_value, v_return);
}

/* BOOL:VOID (./mx-marshal.list:21) */
void
_mx_marshal_BOOLEAN__VOID (GClosure     *closure,
                           GValue       *return_value G_GNUC_UNUSED,
                           guint         n_param_values,
                           const GValue *param_values,
                           gpointer      invocation_hint G_GNUC_UNUSED,
                           gpointer      marshal_data)
{
  typedef gboolean (*GMarshalFunc_BOOLEAN__VOID) (gpointer     data1,
                                                  gpointer     data2);
  register GMarshalFunc_BOOLEAN__VOID callback;
  register GCClosure *cc = (GCClosure*) closure;
  register gpointer data1, data2;
  gboolean v_return;

  g_return_if_fail (return_value != NULL);
  g_return_if_fail (n_param_values == 1);

  if (G_CCLOSURE_SWAP_DATA (closure))
    {
      data1 = closure->data;
      data2 = g_value_peek_pointer (param_values + 0);
    }
  else
    {
      data1 = g_value_peek_pointer (param_values + 0);
      data2 = closure->data;
    }
  callback = (GMarshalFunc_BOOLEAN__VOID) (marshal_data ? marshal_data : cc->callback);

  v_return = callback (data1,
                       data2);

  g_value_set_boolean (return_value, v_return);
}

